src/mlpack/core/cereal/pointer_vector_wrapper.hpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file core/cereal/pointer_vector_wrapper.hpp
+ * @author Omar Shrit
+ *
+ * Implementation of a vector wrapper to enable the serialization of
+ * the size of the vector in cereal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CEREAL_POINTER_VECTOR_WRAPPER_HPP
+#define MLPACK_CORE_CEREAL_POINTER_VECTOR_WRAPPER_HPP
+
+#include <cereal/archives/json.hpp>
+#include <cereal/archives/portable_binary.hpp>
+#include <cereal/archives/xml.hpp>
+#include <cereal/types/vector.hpp>
+
+#include "pointer_wrapper.hpp"
+
+namespace cereal {
+
+template<class T>
+class pointer_vector_wrapper
```

Comment:
Also same comments here about class naming and comment location.  I think this is the last one.  I hope.  But I'll find out when I keep scrolling. :)